framework/DataObjects/HistorySet.py

```diff
@@ -13,618 +12,290 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """
-Created on Feb 16, 2013
-
-@author: alfoa
+  Specialized implementation of DataObject for data with single-valued inputs and outputs that share
+  a single common pivot parameter in the outputs, for each realization.
 """
-#for future compatibility with Python 3--------------------------------------------------------------
+#For future compatibility with Python 3
 from __future__ import division, print_function, unicode_literals, absolute_import
 import warnings
 warnings.simplefilter('default',DeprecationWarning)
-if not 'xrange' in dir(__builtins__):
-  xrange = range
-#End compatibility block for Python 3----------------------------------------------------------------
 
-#External Modules------------------------------------------------------------------------------------
+import sys,os
+import __builtin__
+import functools
 import copy
+import cPickle as pk
 import itertools
-import numpy as np
-import os
-from scipy import spatial
-#External Modules End--------------------------------------------------------------------------------
+import xml.etree.ElementTree as ET
 
-#Internal Modules------------------------------------------------------------------------------------
-from utils.cached_ndarray import c1darray
-from .Data import Data, NotConsistentData, ConstructError
-from utils import utils,mathUtils
-import Files
-#Internal Modules End--------------------------------------------------------------------------------
+import abc
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+from BaseClasses import BaseType
+from Files import StaticXMLOutput
+from utils import utils, cached_ndarray, InputData, xmlUtils, mathUtils
+try:
+  from .DataSet import DataSet
+except ValueError: #attempted relative import in non-package
+  from DataSet import DataSet
+
+# for profiling with kernprof
+try:
+  __builtin__.profile
+except AttributeError:
+  # profiler not preset, so pass through
+  def profile(func):
```

Comment:
This try/except seems off. If \_\_builtin\_\_.profile exists, then it still needs to be called as \_\_builtin\_\_.profile, while if it doens't, profile will exist in the main name space. If profile is already in the main name space from \_\_builtin\_\_, why bother importing \_\_builtin\_\_ to test it instead of just testing the one in the main name space? Am I missing something?

Reply:
See the accepted answer to [this inquiry](https://stackoverflow.com/questions/18229628/python-profiling-using-line-profiler-clever-way-to-remove-profile-statements), where I got it from.

Reply:
What is in that answer is exactly what I would expect here, but isn't what *is* here. Either the try needs to read "profile = \_\_builtin\_\_.profile", or the except needs to have "\_\_builtin\_\_.profile = profile". As written, profile is in one of two different namespaces depending on whether the try fails or not.

Reply:
I see.  Fixed using your first suggestion.